drivers/hmis/app/models/hmis/hud/custom_case_note.rb

```diff
@@ -0,0 +1,24 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+class Hmis::Hud::CustomCaseNote < Hmis::Hud::Base
+  self.table_name = :CustomCaseNote
+  self.sequence_name = "public.\"#{table_name}_id_seq\""
+
+  belongs_to :client, **hmis_relation(:PersonalID, 'Client')
+  belongs_to :enrollment, **hmis_relation(:EnrollmentID, 'Enrollment')
+  belongs_to :user, **hmis_relation(:UserID, 'User')
+  belongs_to :data_source, class_name: 'GrdaWarehouse::DataSource'
+  alias_to_underscore [:CustomCaseNoteID, :PersonalID, :EnrollmentID]
+
+  before_create do |record|
+    record.CustomCaseNoteID ||= Hmis::Hud::Base.generate_uuid
+  end
```

Comment:
this can happen with https://github.com/greenriver/hmis-warehouse/blob/c5db99717eab28e20611d97caabb5eb97ba46c46/drivers/hmis/app/models/hmis/hud/base.rb#L63-L68  if you set `hud_key` on the class

Reply:
good to know, I'll change it. Thanks!